R/filter.R

```diff
@@ -1,17 +1,108 @@
-#' Keep rows that match a condition
+#' Keep or drop rows that match a condition
 #'
-#' The `filter()` function is used to subset a data frame,
-#' retaining all rows that satisfy your conditions.
-#' To be retained, the row must produce a value of `TRUE` for all conditions.
-#' Note that when a condition evaluates to `NA`
-#' the row will be dropped, unlike base subsetting with `[`.
+#' @description
+#' These functions are used to subset a data frame, applying the expressions in
+#' `...` to determine which rows should be kept or dropped.
 #'
-#' The `filter()` function is used to subset the rows of
-#' `.data`, applying the expressions in `...` to the column values to determine which
-#' rows should be retained. It can be applied to both grouped and ungrouped data (see [group_by()] and
-#' [ungroup()]). However, dplyr is not yet smart enough to optimise the filtering
-#' operation on grouped datasets that do not need grouped calculations. For this
-#' reason, filtering is often considerably faster on ungrouped data.
+#' - `filter()` _keeps_ rows where the conditions evaluate to `TRUE`.
+#'
+#' - `r lifecycle::badge("experimental")` `filter_out()` _drops_ rows where the
+#'   conditions evaluate to `TRUE`.
+#'
+#' Multiple conditions can be supplied separated by a comma. These will be
+#' combined with the `&` operator.
+#'
+#' Both `filter()` and `filter_out()` treat `NA` like `FALSE`. This subtle
+#' behavior can impact how you write your conditions when missing values are
+#' involved. See the section on `Missing values` for important details and
+#' examples.
+#'
+#' @inheritParams arrange
+#' @inheritParams args_by
+#'
+#' @param ... <[`data-masking`][rlang::args_data_masking]> Expressions that
+#'   return a logical value, and are defined in terms of the variables in
+#'   `.data`. If multiple expressions are included, they are combined with the
+#'   `&` operator. Only rows for which all conditions evaluate to `TRUE` are
+#'   kept (for `filter()`) or dropped (for `filter_out()`).
+#'
+#' @param .preserve Relevant when the `.data` input is grouped. If `.preserve =
+#'   FALSE` (the default), the grouping structure is recalculated based on the
+#'   resulting data, otherwise the grouping is kept as is.
+#'
+#' @returns
+#' An object of the same type as `.data`. The output has the following
+#' properties:
+#'
+#' * Rows are a subset of the input, but appear in the same order.
+#' * Columns are not modified.
+#' * The number of groups may be reduced (if `.preserve` is not `TRUE`).
+#' * Data frame attributes are preserved.
+#'
+#' @section Missing values:
```

Comment:
```suggestion
#' ## Missing values:
```

And move up in to the description? I don't know if we have an official position on this style.

Reply:
Given all the other sections below, probably not worth it.